src/marks/contour.js

```diff
@@ -0,0 +1,113 @@
+      if (valueType === "string" || valueType === "function") {
+        this.channels[key].value = {transform: () => [], defer: value};
+      }
+    }
+  }
+  render(index, scales, channels, dimensions, context) {
+    const {geometry: G} = channels;
+    const path = geoPath();
+    return create("svg:g", context)
+      .call(applyIndirectStyles, this, dimensions, context)
+      .call(applyTransform, this, scales)
+      .call((g) => {
+        g.selectAll()
+          .data(index)
+          .enter()
+          .append("path")
+          .call(applyDirectStyles, this)
+          .attr("d", (i) => path(G[i]))
+          .call(applyChannelStyles, this, channels);
+      })
+      .node();
+  }
+}
+
+function contourGeometry(options) {
+  const {thresholds = thresholdSturges} = options; // TODO thresholdAuto; match density mark
+  return initializer(options, function (data, facets, channels, scales, dimensions, context) {
+    let {x1, y1, x2, y2} = channels;
+    ({x: [x1], y: [y1]} = Position({x: x1, y: y1}, scales, context)); // prettier-ignore
+    ({x: [x2], y: [y2]} = Position({x: x2, y: y2}, scales, context)); // prettier-ignore
+    let V = channels.value.value;
+    const dx = x2 - x1;
+    const dy = y2 - y1;
+    const {pixelSize: k, width = Math.round(Math.abs(dx) / k), height = Math.round(Math.abs(dy) / k)} = this;
+
+    // Interpolate the raster grid, as needed.
+    if (this.interpolate) {
+      const kx = width / Math.abs(dx);
+      const ky = height / Math.abs(dy);
+      const {x: X, y: Y} = Position(channels, scales, context);
+      const IX = X && map(X, (x) => (x - x1) * kx, Float64Array);
+      const IY = Y && map(Y, (y) => (y - y1) * ky, Float64Array);
+      const index = this.filter(facets[0], {x: channels.x, y: channels.y}, {x: IX, y: IY});
+      V = this.interpolate(index, width, height, IX, IY, V); // TODO faceting?
+    }
+
+    // Blur the raster grid, if desired.
+    if (this.blur > 0) blur2({data: V, width, height}, this.blur);
+
+    // Compute the contours.
+    const geometries = contours().thresholds(thresholds).size([width, height])(V);
+
+    // Rescale the contour multipolygon from grid to screen coordinates.
+    for (const {coordinates} of geometries) {
+      for (const rings of coordinates) {
+        for (const ring of rings) {
+          for (const point of ring) {
+            point[0] = (point[0] / width) * dx + x1;
+            point[1] = (point[1] / height) * dy + y1;
+          }
+        }
+      }
+    }
+
+    // Compute any deferred channels. X and Y are set to as many zeroes as there
+    // are geometries, in order to pass the filter.
+    const newChannels = {geometry: {value: geometries}};
+    for (const key in this.channels) {
+      const value = this.channels[key].value;
+      if (value.defer) newChannels[key] = {value: valueof(geometries, value.defer), scale: true};
+      else if (key === "x" || key === "y") newChannels[key] = {value: valueof(geometries, 0)};
```

Comment:
You can set the filter option on the channel to disable filtering. I’ll fix this.

Reply:
I was able to reproduce the filtering problem using Plot.contour instead of Plot.raster in the vapor equal-earth barycentric interpolate test. The latest commit fixes it.

Reply:
Oops, one more fix needed…

<img width="968" alt="Screenshot 2023-01-09 at 11 01 46 AM" src="https://user-images.githubusercontent.com/230541/211387064-0201e642-7e57-4feb-9a06-178f47161d28.png">

```js
Plot.contour(await vapor(), {
  fill: (d) => d,
  x: (d, i) => (i % 360) - 180 + 0.5,
  y: (d, i) => 90 - ((i / 360) | 0) + 0.5,
  blur: 0.5,
  thresholds: 20,
  stroke: "currentColor",
  strokeWidth: 0.5,
  interpolate: "barycentric",
  clip: "sphere"
})
```